src/Components/SignIn.js

```diff
@@ -13,22 +15,58 @@ const validation = Yup.object().shape({
     .required('Must enter a password')
 })
 
+const SIGN_IN = gql`
+  mutation sign_in($data: SigninInput!){
+    signin(data: $data){
+      user{
+        id
+      }
+      jwt
+    }
+  }
+`
+
 export default function SignIn () {
+  const [error, setError] = useState(null)
+  const history = useHistory()
+  const [signin] = useMutation(SIGN_IN, {
+    onCompleted: LoginSuccess,
+    onError: LoginFailed
+  })
+  function LoginSuccess ({ signin }) {
+    Storage.setToken(signin.jwt)
+    setError(false)
+    history.push('/home')
+  }
+
+  function LoginFailed () {
+    return (
+      setError(true)
+    )
+  }
+
   return (
     <div className='outer'>
       <div className='middle'>
         <div className='inner'>
           <div className='wrapper fadeInDown'>
             <div id='formContent'>
               <h1>Sign In</h1>
+              {error &&
+                <h3 className='error'>  oh snap! Something went wrong  </h3>}
+              {error === false &&
+                <h3 className='correct'>  Log in successfully </h3>}
               <Formik
                 initialValues={{ username: '', password: '' }}
                 validationSchema={validation}
-                onSubmit={(values, { setSubmitting, resetForm }) => {
+                onSubmit={async (values, { setSubmitting }) => {
                   setSubmitting(true)
-                  alert(JSON.stringify({ values }))
-                  resetForm()
-                  setSubmitting(false)
+                  await
+                  signin({
```

Comment:
Place the thing you're awaiting right after the `await` keyword:

```suggestion
                  await signin({
```